presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -700,7 +701,12 @@ protected Type visitTimeLiteral(TimeLiteral node, StackableAstVisitorContext<Con
         protected Type visitTimestampLiteral(TimestampLiteral node, StackableAstVisitorContext<Context> context)
         {
             try {
-                parseTimestampLiteral(session.getTimeZoneKey(), node.getValue());
+                if (SystemSessionProperties.isLegacyTimestamp(session)) {
+                    parseTimestampLiteral(session.getTimeZoneKey(), node.getValue());
+                }
+                else {
+                    parseTimestampLiteral(node.getValue());
+                }
             }
             catch (Exception e) {
                 throw new SemanticException(INVALID_LITERAL, node, "'%s' is not a valid timestamp literal", node.getValue());
```

Comment:
We don't have tests for this and also we need the equivalent for TIME literals. I started to work on this and want to added a quick commit to this PR, but abandoned -- I am not sure `FunctionAssertions` is the best way to test this, since it seems to take different execution path than actual query execution (it seems not to run ExpressionAnalyzer directly :/ )
I didn't create issue for this (someone could pick it up and make a conflict here), so #4 is for the record.

Reply:
Leaving for future improvement